training/unicharset_extractor.cpp

```diff
@@ -50,7 +50,7 @@ static void AddStringsToUnicharset(const GenericVector<STRING>& strings,
                                      /*report_errors*/ true,
                                      strings[i].string(), &normalized)) {
       for (const string& normed : normalized) {
-        if (normed.empty() || IsWhitespace(normed[0])) continue;
+        if (normed.empty() || IsUTF8Whitespace(normed.c_str())) continue;
```

Comment:
The old code tried to check whether the first character of `normed` is a whitespace character.
The new code checks whether `normed` consists only of whitespace characters. If there are two or more characters, that is not the same operation (and it will take more time because not only the first character is examined).

Reply:
`if (normed.empty() || IsUTF8Whitespace(normed[0].c_str())) continue;`

Is that OK?

Reply:
`if (normed.empty() || IsUTF8Whitespace(normed.substr(0,1).c_str())) continue; `

Is that OK?

Any other suggestion?

Reply:
I have printed "normed"，and found it is just a single word read from chi_sim.training_text   every time

Reply:
Did you mean "single character"? Then the new code is fine.

Reply:
yes. I think it is ok. And from the context, I think it just want to strip the whitespace